packages/opentelemetry-id-generator-aws-xray/src/platform/node/AwsXrayIdGenerator.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as crypto from 'crypto';
+import { IdGenerator } from '@opentelemetry/core';
+
+const SPAN_ID_BYTES = 8;
+const TRACE_ID_BYTES = 16;
+const TIME_BYTES = 4;
+
+/** IdGenerator that generates trace IDs conforming to AWS X-Ray format.
+ * https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids
+ */
+export class AwsXRayIdGenerator implements IdGenerator {
+  /**
+   * Returns a random 16-byte trace ID formatted/encoded as a 32 lowercase hex
+   * characters corresponding to 128 bits. The first 4 bytes correspond to the current
+   * time, in seconds, as per X-Ray trace ID format.
+   */
+  generateTraceId(): string {
+    const nowSec = Math.floor(Date.now() / 1000).toString(16);
```

Comment:
Consistently use `toString('hex')` or `toString(16)`

Reply:
I use ```16``` rather than ```'hex'``` here is because this part we transfer a ```number``` to ```String``` while the other one is transferring ```Byte``` to ```String```. The build-in ```toString()``` methods are different from each other.